src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Analog;
+using System.Text;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Iot.Device.Spi;
+
+#pragma warning disable CS1591
+
+namespace Iot.Device.Arduino
+{
+    /// <summary>
+    /// Implements an interface to an arduino board which is running Firmata.
+    /// See documentation on how to prepare your arduino board to work with this.
+    /// Note that the program will run on the PC, so you cannot disconnect the
+    /// Arduino while this driver is connected.
+    /// </summary>
+    public class ArduinoBoard : IDisposable
+    {
+        private Stream _serialPortStream;
+        private FirmataDevice _firmata;
+        private Version _firmwareVersion;
+        private string _firmwareName;
+        private List<SupportedPinConfiguration> _supportedPinConfigurations;
+
+        public ArduinoBoard(Stream serialPortStream)
```

Comment:
Should we also allow constructor overloads that take in either a SerialPort object or just an int for the port that is using?

Reply:
I was thinking of doing that, yes, but I wanted to ask you whether it's ok to include a reference to System.IO.Ports. But now I see that this is part of Iot.Device.Bindings already. 
I'll add an overload that takes a port name (on linux, it may be /dev/ttyUSBx or /dev/serialx or /dev/ttyAMAx, so a number is not enough).

Reply:
In the other bindings, it usually just take a string and then manage the serial port internally as yes, it can be any string.

Reply:
I prefer to keep the Stream interface as well, because it's more generic. Maybe one wants to forward this over a network or replay data from a file.

Reply:
Yes, sure, keep the stream and add the string to open as serial port then. And yes, some implementation exists over network.